src/instana/instrumentation/boto3_inst.py

```diff
@@ -5,142 +5,182 @@
 import json
 import wrapt
 import inspect
+from typing import TYPE_CHECKING, Any, Callable, Dict, Tuple, Sequence, Type
+from opentelemetry.semconv.trace import SpanAttributes
 
-from ..log import logger
-from ..singletons import tracer, agent
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.singletons import tracer, agent
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.propagators.format import Format
+from instana.span.span import get_current_span
+
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
+    from botocore.auth import SigV4Auth
+    from botocore.client import BaseClient
 
 try:
-    import opentracing as ot
     import boto3
     from boto3.s3 import inject
 
-    def extract_custom_headers(span, headers):
+    def extract_custom_headers(span: "InstanaSpan", headers: Dict[str, Any]) -> None:
         if agent.options.extra_http_headers is None or headers is None:
```

Comment:
If the `headers` can be `None`, we should update the type hint to `Optional[Dict[str, Any]] = None` and also update the if:
```suggestion
    def extract_custom_headers(span: "InstanaSpan", headers: Optional[Dict[str, Any]] = None) -> None:
        if not agent.options.extra_http_headers or not headers:
```

Reply:
Done.